<?xml version="1.0"?>

<launch>
	<!-- see cfg/mono.cfg for detail -->
  <arg name="left_cam_serial"             default="4103423533" />
  <arg name="right_cam_serial"            default="4103423537" />
	<arg name="namespace"                   default="camera" />
	<arg name="auto_exposure"	              default="true" />
	<arg name="auto_gain"		                default="false" />
	<arg name="color"		                    default="0" /> <!-- 0:MONO8, 1:MONO16, 2:YUV, 3:YCbCr, 4:BGR5, 5:BGR565, 6:BGR8, 7:BGRA8, 9:BGRY8, 9:RGB8, 10:RGBA8, 11, RGBY8 -->
	<arg name="config_path"		              default="" />
	<arg name="exposure_time"	              default="85" /> <!-- 85 -->
	<arg name="force_streaming"	            default="false" />
	<arg name="l_frame_id"		              default="/left_camera" />
	<arg name="r_frame_id"		              default="/right_camera" />
	<arg name="frame_rate"		              default="236" /> <!-- 30 -->
	<arg name="gain"		                    default="2000" /> <!-- 100 -->
	<arg name="gain_boost"		              default="false" />
	<arg name="hardware_gamma"	            default="false" />
	<arg name="pixel_clock"		              default="480" /> <!-- 86 -->
	<arg name="extended_pixel_clock"		    default="true" />
	<arg name="trigger"		                  default="0" /> <!-- 0:AUTO, 1:SOFTWARE, 2:HARDWARE_RISING, 3:HARDWARE_FALLING -->
	<arg name="zoom"		                    default="1" />
	<arg name="height"		                  default="840" />
	<arg name="width"		                    default="1936" />
	<arg name="height_offset"		            default="0" />
	<arg name="width_offset"		            default="0" />
	<arg name="brightness_AOI_Height"		    default="400" />
  <arg name="brightness_AOI_Width"		    default="1936" />
  <arg name="brightness_AOI_X"		        default="0" />
  <arg name="brightness_AOI_Y"            default="0" />
  <arg name="software_binning"		        default="True" />
  <arg name="visualize_brightness_AOI"    default="False" />
  <arg name="auto_exposure_hysteresis"		default="2" />
  <arg name="auto_exposure_max"		        default="4.2" />
  <arg name="auto_exposure_reference"		  default="80" />
  <arg name="auto_exposure_skip_frames"	  default="20" />
  <arg name="auto_exposure_speed"		      default="50" />

	<group ns="$(arg namespace)" >
		<node pkg="nodelet" type="nodelet" name="ueye_manager" args="manager" output="screen"/>

		<node pkg="nodelet" type="nodelet" name="camera_nodelet" args="load ueye/StereoNodelet ueye_manager" >
		  <param name="lSerialNo" type="str"        value="$(arg left_cam_serial)" />
		  <param name="rSerialNo" type="str"        value="$(arg right_cam_serial)" />
		  <param name="auto_gain"		                value="$(arg auto_gain)" />
		  <param name="color"		                    value="$(arg color)" />
		  <param name="config_path"	                value="$(arg config_path)" />
		  <param name="exposure_time"	              value="$(arg exposure_time)" />
		  <param name="auto_exposure"	              value="$(arg auto_exposure)" />
		  <param name="force_streaming" 	          value="$(arg force_streaming)" />
		  <param name="l_frame_id"		              value="$(arg l_frame_id)" />
		  <param name="r_frame_id"		              value="$(arg r_frame_id)" />
		  <param name="frame_rate"	                value="$(arg frame_rate)" />
		  <param name="gain"		                    value="$(arg gain)" />
		  <param name="gain_boost"	                value="$(arg gain_boost)" />
		  <param name="hardware_gamma"	            value="$(arg hardware_gamma)" />
		  <param name="pixel_clock"	                value="$(arg pixel_clock)" />
		  <param name="extended_pixel_clock"	      value="$(arg extended_pixel_clock)" />
		  <param name="trigger"		                  value="$(arg trigger)" />
		  <param name="zoom"		                    value="$(arg zoom)" />
		  <param name="l_AOI_Height"		            value="$(arg height)" />
		  <param name="l_AOI_Width"		              value="$(arg width)" />
		  <param name="l_AOI_Y"		                  value="$(arg height_offset)" />
		  <param name="l_AOI_X"		                  value="$(arg width_offset)" />
		  <param name="r_AOI_Height"		            value="$(arg height)" />
		  <param name="r_AOI_Width"		              value="$(arg width)" />
		  <param name="r_AOI_Y"		                  value="$(arg height_offset)" />
		  <param name="r_AOI_X"		                  value="$(arg width_offset)" />
		  <param name="l_brightness_AOI_Height"		  value="$(arg brightness_AOI_Height)" />
      <param name="l_brightness_AOI_Width"		  value="$(arg brightness_AOI_Width)" />
      <param name="l_brightness_AOI_X"	      	value="$(arg brightness_AOI_X)" />
      <param name="l_brightness_AOI_Y"	      	value="$(arg brightness_AOI_Y)" />
		  <param name="r_brightness_AOI_Height"	  	value="$(arg brightness_AOI_Height)" />
      <param name="r_brightness_AOI_Width"		  value="$(arg brightness_AOI_Width)" />
      <param name="r_brightness_AOI_X"		      value="$(arg brightness_AOI_X)" />
      <param name="r_brightness_AOI_Y"		      value="$(arg brightness_AOI_Y)" />
      <param name="software_binning"		        value="$(arg software_binning)" />
      <param name="visualize_brightness_AOI"		value="$(arg visualize_brightness_AOI)" />
      <param name="auto_exposure_hysteresis"		value="$(arg auto_exposure_hysteresis)" />
      <param name="auto_exposure_max"		        value="$(arg auto_exposure_max)" />
      <param name="auto_exposure_reference"		  value="$(arg auto_exposure_reference)" />
      <param name="auto_exposure_skip_frames"		value="$(arg auto_exposure_skip_frames)" />
      <param name="auto_exposure_speed"		      value="$(arg auto_exposure_speed)" />
      
		</node>
		
		<!-- PRINT THE IMAGE SEQUENCE NUMBER, RESOLUTION AND FRAME RATE -->
		<!--<node pkg="nodelet" type="nodelet" name="framerate_nodelet" args="load ueye/FramerateNodelet ueye_manager"/>-->
	</group>
</launch>
